lib/bundix/source.rb

```diff
@@ -21,6 +26,36 @@ def download(file, url)
       end
     end
 
+    def inject_credentials_from_bundle_config(uri)
+      @credentials ||= bundle_config_credentials
+      key = "BUNDLE_#{uri.host.upcase.gsub(/\./, '__')}" 
+      user, password = @credentials[key]
+      if user
+        uri.user = user
+        uri.password = password
+      end
+    end
+
+    def bundle_config_credentials
+      credentials = {}
+
+      files = [
+        File.expand_path("~/.bundle/config"),
+        "./.bundle/config",
+      ]
+
+      files.each do |file|
+        next unless File.exist?(file)
+        File.read(file)
+          .scan(/^(?<key>BUNDLE_.*?):\s*(?<username>.*)(?::(?<password>.*))?/)
```

Comment:
I wasn't sure whether this project wanted to rely on YAML being available so I didn't do this "correctly", but happy to revise to actually parse the file if that would be preferable.